semantics/src/traversal/expressions.rs

```diff
@@ -611,12 +665,22 @@ fn expr_attribute_call_type(
     exp: &Spanned<fe::Expr>,
 ) -> Result<CallType, SemanticError> {
     if let fe::Expr::Attribute { value, attr } = &exp.node {
-        return match value.node {
-            fe::Expr::Name(builtins::SELF) => Ok(CallType::SelfAttribute {
-                func_name: attr.node.to_string(),
-            }),
-            _ => Ok(CallType::ValueAttribute),
+        if let fe::Expr::Name(name) = value.node {
+            use builtins::Object;
+            match Object::from_str(name) {
+                Ok(Object::Block) => todo!(),
+                Ok(Object::Chain) => todo!(),
+                Ok(Object::Msg) => todo!(), // TODO: error because msg has no methods?
```

Comment:
Yeah, we could use the `UndefinedValue` error here or create something more specific like `UndefinedFunction`.

Reply:
I made these UndefinedValues for now.